src/app/libp2p_helper/src/codanet.go

```diff
@@ -237,11 +239,56 @@ type Helper struct {
 	GatingState       *CodaGatingState
 	ConnectionManager *CodaConnectionManager
 	BandwidthCounter  *metrics.BandwidthCounter
+	MsgStats          *MessageStats
 	Seeds             []peer.AddrInfo
 	NodeStatus        string
 	pxDiscoveries     chan peer.AddrInfo
 }
 
+type MessageStats struct {
+	min   uint64
+	avg   uint64
+	max   uint64
+	total uint64
+	sync.RWMutex
+}
+
+func (ms *MessageStats) UpdateMetrics(val uint64) {
+	ms.Lock()
+	defer ms.Unlock()
+	if ms.max < val {
+		ms.max = val
+	}
+
+	if ms.min > val {
+		ms.min = val
+	}
+
+	ms.total++
+	if ms.avg == 0 {
+		ms.avg = val
+	} else {
+		ms.avg = (ms.avg*(ms.total-1) + val) / ms.total
```

Comment:
I think you may lose a little precision when storing avg/total instead of sum/total -- sum/total then you just always make `avg` computed from sum/total whenever you need it. Unless you think sum will overflow -- in that case, definitely keep it the way it is.

Reply:
Actually, I suppose it would overflow as written too since the multiplication is before the division.

Reply:
How much precision do we actually lose for every `n * c / c`? Is it actually enough to warrant changing this?

Reply:
It's not like we depend on this having extreme precision, wouldn't think that the precision loss would be significant past a few decimal places.